TabulateSmarterTestAdminPackage.Common/Validators/ValidatorCollection.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace TabulateSmarterTestAdminPackage.Common.Validators
+{
+    public class ValidatorCollection
+    {
+        private readonly Queue<Validator> _validators;
+
+        public ValidatorCollection()
+        {
+            _validators = new Queue<Validator>();
+        }
+
+        public ValidatorCollection(Validator validator)
+        {
+            _validators = new Queue<Validator>();
+            _validators.Enqueue(validator);
+        }
+
+        public ValidatorCollection(IEnumerable<Validator> validators)
+        {
+            _validators = new Queue<Validator>();
+            Add(validators);
+        }
+
+        public ValidatorCollection Add(Validator validator)
+        {
+            _validators.Enqueue(validator);
+            return this;
+        }
+
+        public ValidatorCollection Add(IEnumerable<Validator> validators)
+        {
+            foreach (var validator in validators)
+            {
+                _validators.Enqueue(validator);
+            }
+            return this;
+        }
+
+        public bool ObjectPassesValidation(object value)
+        {
+            return _validators.All(x => x.IsValid(value));
+        }
+
+        public string ObjectValidationErrors(object value)
+        {
+            return _validators
+                .Where(x => !x.IsValid(value))
+                .Select(x => x.GetMessage())
+                .Aggregate((i, j) => i + j);
```

Comment:
Do we need to aggregate the validation messages?  For example, with a Required String check and a Minimum Length check (maybe not a realistic example, but to show a point) would we want them both there or is it enough to return the first one that fails?

If we need all of them, I don't think this Aggregate would comma separate them.  I think you'd want:

`String.Join(", ", this.Where(x => !x.IsValid(value)).Select(x => x.GetMessage()).ToArray());`